src/olympia/discovery/views.py

```diff
@@ -22,6 +22,11 @@ def get_queryset(self):
         addons = self.queryset.in_bulk(ids)
 
         # Patch items to add addons.
+        result = []
         for item in discopane_items:
-            item.addon = addons[item.addon_id]
-        return discopane_items
+            try:
+                item.addon = addons[item.addon_id]
+                result.append(item)
+            except KeyError:
+                pass
```

Comment:
I thought about it, but since this API is going to be hit a lot... Maybe it'll just result in noise. If something is missing, I assume the editorial team will see it and file a bug for us.